### ISSUE TITLE
Using field maps

### ISSUE THREAD

**Issue by cchaswell** [HALFpipe/HALFpipe#177]
**What happened?**
I get an error when setting up the pipeline with field maps. This is my first time using field maps. It seems like the Phase Encoding direction is wrong for me. 

 e = ValueError('At least one of the EPI runs with alternative ph
│               ase-encoding blips is missing the required "PhaseEncodingDir
│               ection" metadata entry.')
**Is there an error message?**
Please copy it here. If you are unsure, please provide the contents of the `err.txt` file from your `Halfpipe` working directory. Note that this log file may contain filenames from your computer.

**Which settings did you use?**
Maybe I am doing this incorrectly. I have 2 EPI field maps (and A-P and P-A). I enter the field maps as {subject}_{direction}.nii.gz. The program finds the 2 field maps per subject. 

Am I doing this correctly or do I need to enter it as 2 separate maps? I also believe that the phase encoding direction is P-A on my functional data but am not sure how to confirm.

**Comment by HippocampusGirl** [comment]
It sounds like you are using it correctly. Could you send me the log file and which version you are using?

**Comment by cchaswell** [comment]
[log.txt](https://github.com/HALFpipe/HALFpipe/files/7008193/log.txt)
Halfpipe version 1.0.0b6+40.g8199f57

**Comment by cchaswell** [comment]
GE doesn’t put anything in the metadata that just tells you the PE direction, but I did confirm with our experts that it is P-A direction.

**Comment by cchaswell** [comment]
My cluster admin did ask about this:

"In fmriprep when you specify fieldmaps you have to have all the phase encode info inside the json file and in the JSON of the fieldmaps it has to have the “IntendedFor” field to specify which functionals to correct.  How is your pipeline figuring out that part?"

The pipeline worked fine with those scans when I wasn't using the field map option.

**Comment by HippocampusGirl** [comment]
- The version you are using is more than half a year old. I greatly improved the handling of field maps for the upcoming release. Could you try again with the current `latest` version? You can download it from <https://download.fmri.science/singularity/halfpipe-halfpipe-latest.simg>

- Regarding the assignment of field maps, this is how we described it in the paper:
   > A further challenge is generating database queries that flexibly interface with the logic of neuroimaging and processing pipelines, which is relevant in the context of missing data. HALFpipe always tries to execute the best possible processing pipeline based on the data that is available. For example, a field map may have been routinely acquired before each functional scan in a particular dataset. If one of these field maps is missing, HALFpipe flexibly assigns another field map, for example one belonging to the preceding functional scan. However, HALFpipe will not use a field map from another scan session, as field inhomogeneities are likely to have changed. Finally, HALFpipe does not fail if a field map is missing, but simply omits the distortion correction step for that subject. Other examples include the ability of HALFpipe to match structural to functional images, and match task events to a functional scan. This strategy is used throughout the construction of processing workflows.
  
  In other words, HALFpipe generates the `IntendedFor` flexibly.

**Comment by cchaswell** [comment]
That is what I get for keeping old singularity images in my folder. I had a newer version from May that I should have been using. 

I updated to the latest image and it looks like the field map stuff worked properly in the setup. At the end of setup I did get an error about FS license.

[2021-08-23 09:50:59,0676] [nipype.workflow ] [IMPORTANT] Slice-timing correction will be included.
[2021-08-23 09:51:03,0183] [nipype.workflow ] [IMPORTANT] Slice-timing correction will be included.
[2021-08-23 09:51:16,0464] [halfpipe        ] [ERROR    ] fMRIPrep needs to use FreeSurfer commands, but a valid license file for FreeSurfer could not be found. 
│ Halfpipe looked for an existing license file at several paths, in this order: 
│ 1) a "license.txt" file in your Halfpipe working directory 
│ 2) command line argument "--fs-license-file" 
└─Get it (for free) by registering at https://surfer.nmr.mgh.harvard.edu/registration.html
